src/sentry/static/sentry/app/views/settings/components/forms/radioField.jsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styled from 'react-emotion';
+import PropTypes from 'prop-types';
+
+import InputField from './inputField';
+import {growIn} from './styled/animations';
+
+class RadioField extends InputField {
+  static propTypes = {
+    id: PropTypes.number,
+    value: PropTypes.string,
+  };
+
+  onChange = (value, onChange, onBlur, e) => {
+    onChange(value, e);
+    onBlur(value, e);
+  };
+
+  isSelected = ({value, id}) => {
+    return value ? value === id : id === 0;
+  };
+
+  render() {
+    return (
+      <InputField
+        {...this.props}
+        field={({onChange, onBlur, value, disabled, ...props}) => (
+          <div>
+            {(props.choices() || []).map((choice, index) => {
+              const {id, name} = choice;
+              return (
+                <RadioLineItem
+                  key={id}
+                  onClick={this.onChange.bind(this, id, onChange, onBlur)}
+                  index={index}
+                >
+                  <RadioLineButton>
```

Comment:
Any ideas what we need to do to make these accessible?

Reply:
Great question lol. The truth is that the true best thing here would be to use actual <input type="radio"/> elements, because those are the things that screen readers, tab focuses, etc. will 100% always pick up. I'm not against this option, I simply opted for a design that matched the direction we were already heading with the boolean switches.

Another option, if we want to keep this style is to add role and aria attributes to the divs. This is what we are doing on the boolean switches.

A third option is to use visually hidden form elements that match whatever the javascript elements show. This could allows us to have our cake and eat it too, but the complexity might also backfire and cause unknowable craziness.

Maybe @ckj could weight in here?

Reply:
Really happy this is something we're pausing to think about. `aria` and `role` would be a good start. I think we could/should go a step further and have these (and our switches as well) be button-based instead of div-based so that we get tabbing, arrow key functionality, and other browser built-in accessibility features by default.